File: Add Binary/Add Binary.cpp
class Solution {
public:
    string addBinary(string a, string b) {
        int m = a.size();
        int n = b.size();
        int len = max(m, n);
        int shift = 0;
        int res = 0;
        string s = "";
        for (int i = 0; i < len; ++i) {
            --m;
            --n;
            if (m >= 0 && n >= 0) {
                res = a[m] + b[n] + shift - '0' - '0';
            } else if (m >= 0) {
                res = a[m] - '0' + shift;
            } else {
                res = b[n] - '0' + shift;
            }
            shift = res / 2;
            res = res % 2;
            s = to_string(res) + s;
        }
        if (shift != 0) s = to_string(shift) + s;
        return s;
    }
};

File: Add Binary/Add Binary.java
public class Solution {
    public String addBinary(String a, String b) {
        int m = a.length();
        int n = b.length();
        int len = Math.max(m, n);
        int shift = 0;
        int res = 0;
        StringBuilder sb = new StringBuilder();
        for (int i = 0; i < len; ++i) {
            --m;
            --n;
            if (m >= 0 && n >= 0) {
                res = a.charAt(m) + b.charAt(n) + shift - '0' - '0';
            } else if (m >= 0) {
                res = a.charAt(m) - '0' + shift;
            } else {
                res = b.charAt(n) - '0' + shift;
            }
            shift = res / 2;
            res = res % 2;
            sb.insert(0, res);
        }
        if (shift != 0) sb.insert(0, shift);
        return sb.toString();
    }
}